assembly_uploader/release_study.py

```diff
@@ -0,0 +1,79 @@
+import argparse
+import logging
+import tempfile
+import xml.etree.ElementTree as ET
+from pathlib import Path
+from xml.dom import minidom
+
+import requests
+
+from assembly_uploader.submit_study import DROPBOX_DEV, DROPBOX_PROD
+from assembly_uploader.webin_utils import (
+    ensure_webin_credentials_exist,
+    get_webin_credentials,
+)
+
+
+class StudyReleaseError(Exception):
+    pass
+
+
+def release_study(accession: str, is_test: bool = False, xml_path: Path = None):
+    """
+    Immediately publicly release a study, so that it is no longer private/embargoed/held in ENA.
+    :param accession: study accession
+    :param is_test: If true, will use wwwdev ENA Test Webin dropbox
+    :param xml_path: Optional path to use for the submission xml file, otherwise a temp dir will be used.
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        xml_path = xml_path or Path(tmp_dir) / Path(accession + ".xml")
+
+        with xml_path.open("wb") as release_file:
+            submission = ET.Element("SUBMISSION")
+            actions = ET.SubElement(submission, "ACTIONS")
+            action_sub = ET.SubElement(actions, "ACTION")
+            release = ET.SubElement(action_sub, "RELEASE")
+            release.set("target", accession)
+            dom = minidom.parseString(ET.tostring(submission, encoding="utf-8"))
+            release_file.write(dom.toprettyxml().encode("utf-8"))
+
+        endpoint = DROPBOX_DEV if is_test else DROPBOX_PROD
+
+        files = {
+            "SUBMISSION": xml_path.open("rb"),
+        }
+
+        release_report = requests.post(
+            endpoint, files=files, auth=get_webin_credentials()
+        )
+        receipt_xml_str = release_report.content.decode("utf-8")
+        if 'success="true"' in receipt_xml_str:
```

Comment:
Shouldn't we parse the resuling XML, in case it contains an erro code hidden in it (or stuff like that)?

Reply:
Also, can't you ask for the result to be a json? Content-type like?

Reply:
Thanks @mberacochea . The answer to both of these seems to be to switch to the webin V2 API, which supports JSON and a [structured receipt](https://ena-docs.readthedocs.io/en/latest/submit/general-guide/programmatic.html#submission-receipt)

I will do another (future) PR with a wholesale switch to the V2 API and JSON.